core/src/main/scala/filodb.core/memstore/MemStore.scala

```diff
@@ -141,6 +143,9 @@ trait MemStore extends ChunkSource {
    */
   def indexValues(dataset: DatasetRef, shard: Int, indexName: String, topK: Int = 100): Seq[TermInfo]
 
+  def metadata(dataset: DatasetRef, shard: Int, filters: Seq[ColumnFilter],
```

Comment:
1) This is a Trait/interface, detailed ScalaDoc is needed here
2) The name `metadata` is not very descriptive, I have no idea what this method is going to do or what kind of metadata it will return.  I'm guessing maybe something like `queryPartitionKeys`?
3) This will need to be changed per our discussion on the phone to return something like `Seq[Array[Byte]]` where each array of bytes is the partition key data.  

We might want to create a type alias:  `type OnHeapPartitionKey = Array[Byte]` to improve readability.  Check Types.scala, there might be something like that already.  

Another advantage of the new return type, besides much more schema flexibility, is that it will use far less heap.  Map[UTF8String, UTF8String] has HUGE on heap overhead compared to the amount of data it represents - the object overhead is probably 2-3x what the data itself will be.